OpenRA.Platforms.Default/Shader.cs

```diff
@@ -138,10 +139,21 @@ public void PrepareRender()
 			// bind the textures
 			foreach (var kv in textures)
 			{
-				OpenGL.glActiveTexture(OpenGL.GL_TEXTURE0 + kv.Key);
-				OpenGL.glBindTexture(OpenGL.GL_TEXTURE_2D, ((ITextureInternal)kv.Value).ID);
+				var id = ((ITextureInternal)kv.Value).ID;
+
+				// Evict disposed textures from the cache
+				if (OpenGL.glIsTexture(id))
+				{
+					OpenGL.glActiveTexture(OpenGL.GL_TEXTURE0 + kv.Key);
+					OpenGL.glBindTexture(OpenGL.GL_TEXTURE_2D, id);
+				}
+				else
+					unbindTextures.Enqueue(kv.Key);
```

Comment:
Nice catch. Perhaps we should consider throwing `ObjectDisposedExpception` from things like this to help detect use-after-dispose bugs.